File: algorithms/121. Best Time to Buy and Sell Stock/solution.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        minprice = sys.maxsize
        maxprofit = 0
        for p in prices:
            maxprofit = max(maxprofit, p-minprice)
            minprice = min(minprice, p)
        return maxprofit

# 200 / 200 test cases passed.
# Status: Accepted
# Runtime: 43 ms
# 86.49%

File: algorithms/121. Best Time to Buy and Sell Stock/solution.cc
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPrice = INT_MAX, maxProfit = 0;
        for ( int p : prices ) {
            maxProfit = max(maxProfit, p - minPrice);
            minPrice = min(minPrice, p);
        }
        return maxProfit;
    }
};

// 200 / 200 test cases passed.
// Status: Accepted
// Runtime: 6 ms
// beats 45.03%
